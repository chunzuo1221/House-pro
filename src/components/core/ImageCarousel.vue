<template>
  <carousel
    :per-page="1"
    :navigation-enabled="true"
    :pagination-enabled="true"
    pagination-active-color="#528EC1"
    navigation-prev-label="<i class='fa fa-chevron-left'></i>"
    navigation-next-label="<i class='fa fa-chevron-right'></i>"
    :navigation-click-target-size="20"
  >
    <slide v-for="(image, i) in images" :key="i">
      <figure class="image-slide">
        <img :src="image.url">
        <span class="image-name">{{image.name}}</span>
      </figure>
    </slide>
  </carousel>
</template>

<script>
import { Carousel, Slide } from 'vue-carousel'

export default {
  name: 'ImageCarousel',
  components: {
    Carousel,
    Slide
  },
  props: ['images']
}
</script>

<style scoped>
.VueCarousel >>> .VueCarousel-dot-button:focus {
  outline: none !important;
}
.VueCarousel >>> .VueCarousel-pagination {
  position: absolute;
  bottom: 0;
}
.VueCarousel-navigation-prev.VueCarousel-navigation-button {
  left: 50px;
  color: #fff;
  top: calc(50% - 10px);
  font-size: 20px;
}
.VueCarousel-navigation-next.VueCarousel-navigation-button {
  right: 50px;
  color: #fff;
  top: calc(50% - 10px);
  font-size: 20px;
}
.image-slide {
  position: relative;
  min-height: 400px;
  min-width: 600px;
}
.image-slide .image-name {
  border: 2px solid #528EC1;
  padding: 3px 16px;
  position: absolute;
  top: 16px;
  right: 16px;
  color: #528EC1;
  font-weight: bold;
  background: #fff
}
</style>
