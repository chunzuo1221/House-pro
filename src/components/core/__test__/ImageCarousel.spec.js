import { shallowMount } from '@vue/test-utils'
import ImageCarousel from '@/components/core/ImageCarousel'

describe('ImageCarousel', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(ImageCarousel, {
      propsData: {
        images: [
          { url: 'https://images.pexels.com/photos/584399/living-room-couch-interior-room-584399.jpeg', name: 'Living room couch interior room' }
        ]
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
