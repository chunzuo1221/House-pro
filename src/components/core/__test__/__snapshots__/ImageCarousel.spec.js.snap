// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCarousel renders correctly 1`] = `
<carousel-stub
  easing="ease"
  minswipedistance="8"
  mousedrag="true"
  navigateto="0"
  navigationclicktargetsize="20"
  navigationenabled="true"
  navigationnextlabel=<i class='fa fa-chevron-right'></i>
  navigationprevlabel=<i class='fa fa-chevron-left'></i>
  paginationactivecolor="#528EC1"
  paginationcolor="#efefef"
  paginationenabled="true"
  paginationpadding="10"
  paginationsize="10"
  perpage="1"
  resistancecoef="20"
  scrollperpage="true"
  spacepadding="0"
  spacepaddingmaxoffsetfactor="0"
  speed="500"
>
  <slide-stub>
    <figure
      class="image-slide"
    >
      <img
        src="https://images.pexels.com/photos/584399/living-room-couch-interior-room-584399.jpeg"
      />
       
      <span
        class="image-name"
      >
        Living room couch interior room
      </span>
    </figure>
  </slide-stub>
</carousel-stub>
`;
